factory/crypto/cryptoComponentsHandler.go

```diff
@@ -82,15 +82,24 @@ func (mcc *managedCryptoComponents) CheckSubcomponents() error {
 	if check.IfNil(mcc.cryptoComponents.publicKey) {
 		return errors.ErrNilPublicKey
 	}
+	if check.IfNil(mcc.cryptoComponents.p2pPublicKey) {
+		return errors.ErrNilPublicKey
```

Comment:
should use different errors for p2pPublicKey and p2pPrivateKey

Reply:
done, added